doc/backing-store-crd.md

```diff
@@ -178,6 +178,28 @@ spec:
 
 In case the credentials of a backing-store need to be updated due to a periodic security policy or concern, the appropriate secret should be updated by the user, and the operator will be responsible for watching changes in those secrets and propagating the new credential update to the NooBaa system server.
 
+#### BackingStore access management
+
+By default, any BucketClass that is in the same namespace as the BackingStore can refer and use it. For the cases where BackingStore wants to explicitly allow access to BucketClasses in different namespaces, BackingStore can be provided a comma separated namespaces in `metadata.annotations.allowedNamespaces`.
```

Comment:
I'm not sure what's the purpose of adding this annotation. is this for access control? how effective is it, if anyone with permissions to the backingstore NS can change it?

Reply:
Yes, this is for access control. This implementation was under the assumption that org admins will be creating these Backingstore and Namespace stores and will annotate them appropriately while the "sub-projects" will have access to only their own namespace.

Quoting Guy from our discussions on Slack,
> Every org-admin will have its own dedicated odf-mcg deployment on the saas cluster. For every sub-project/team/app inside the org, that needs to use a cloud resource (or any other resource) the org-admin will create the bs/ns resources in the operator namespace (as it is today), but he/she will prepare bucket classes in the sub-project namespace that reference those bs/ns resources by name.
> To restrict sharing bs/ns between sub-projects in the same org, we have two options - set the sub project users rbac to not allow edit access of bucket classes even inside its own namespace, so only org-admin can edit and change, or provide a custom permission annotation on the bs/ns so users cannot use unintended resource (allowedNamespaceSelector or quotas).
> The benefit is that we reduce a lot of complexity of moving the bs/ns resources to app namespace, and only move the “simpler” bucket classes, which will reduce the scope of changes needed in the repo significantly.

Reply:
I think that eventually we will want to disallow any cross NS references (BC can only reference resources in its own NS). I suggest to not add any access control in this PR, especially since we don't have any regression in the current behavior (OBC can reference any bucketclass without access control)

Reply:
Addressed